terps/scott/layouttext.c

```diff
@@ -0,0 +1,94 @@
+//
+//  layouttext.c
+//  scott
+//
+//  Created by Administrator on 2022-01-11.
+//
+
+#include <ctype.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "layouttext.h"
+
+int FindBreak(const char *buf, int pos, int columns)
+{
+
+    int diff = 0;
+
+    while (diff < columns && !isspace(buf[pos])) {
```

Comment:
This is always one of my least-favorite review issues, because it seems so ridiculous, but when passing a `char` to the `ctype` functions, you need to cast it to `unsigned char`.  Why?  Because the C standard says so, that's why...

But it's really to accommodate systems which use macros plus lookup tables to speed up these functions.  Solaris used to do this, and to my recollection NetBSD did as well.  I'm not sure any modern systems still do, but they're allowed to.  On Solaris, `isspace` used to look like this:

```#define	isspace(c)	((__ctype + 1)[c] & _S)```

And `__ctype` is an array mapping `unsigned char` to a set of bits indicating which properties each character has.  The `+ 1` is because EOF (-1 on Solaris and probably ever other existing system) has to be handled as well.  So `isspace(x)` will index directly into the array, and if you have a signed `char` type, and the value is negative, you'll index backward into something that's totally unrelated.

So, long story short, all `ctype` functions are required to  take “an `int`, the value of which shall be representable as an `unsigned char` or shall equal the macro `EOF`.” That's backwards compatibility for you.